factory.go

```diff
@@ -37,11 +38,28 @@ func (f factory) Create(config *Config, logger log.Logger) (deployer.Connector,
 	if err != nil {
 		return &Connector{}, fmt.Errorf("python binary check failed with error: %w", err)
 	}
-	python := cliwrapper.NewCliWrapper(pythonPath, config.WorkDir, logger)
+
+	workdir, err := os.MkdirTemp(filepath.Clean(config.WorkDir), "")
+	if err != nil {
+		return nil, fmt.Errorf(
+			"error creating temporary directory for python connector (%w)", err)
+	}
+
+	pythonFactory, err := cliwrapper.NewCliWrapperFactory(pythonPath, workdir, logger)
+	if err != nil {
+		return nil, err
+	}
+
+	venv, err := cliwrapper.Venv(workdir, pythonPath, logger)
+	if err != nil {
+		return nil, err
+	}
 	return &Connector{
-		config:           config,
-		logger:           logger,
-		pythonCliWrapper: python,
+		config:        config,
+		logger:        logger,
+		pythonFactory: pythonFactory,
+		venv:          venv,
+		lock:          &sync.Mutex{},
 	}, nil
```

Comment:
Again, assuming that this is the only place where a `Connector` is instantiated, it should be defined to _contain_ a `Mutex` and not "reference" it (by instead containing a pointer to it).  That will allow you to instantiate the `Connector` without also (explicitly) instantiating the `Mutex`.  (And, this is a Good Thing™, because there is a one-to-one relationship between the `Connector` and the `Mutex`, so they don't need to be (and shouldn't be) managed separately.)